### DISCUSSION TITLE
Ignoring indexer that doesn't support searching

### DISCUSSION THREAD

**Discussion by ramzh** [cross-seed/cross-seed#354] [Q&A] [answered]
I have set up 3 indexers, 2 public and 1 private, but I keep getting the following: 
```
Configuration file already exists.
info: Validating your configuration...
warn: Ignoring indexer that doesn't support searching: http://192.168.43.11:9696/1/api?apikey=?API
warn: Ignoring indexer that doesn't support searching: http://192.168.43.11:9696/2/api?apikey=?API
warn: Ignoring indexer that doesn't support searching: http://192.168.43.11:9696/12/api?apikey=?API
error: no working indexers available
```
OS: OpenMediaVault Shaitain
Installation: Standalone Docker-Compose/Stack via Portrainer
Network: same network as Prowlarr

**Comment by ramzh** [comment]
Hi

Just saw this as closed. Any reasons why?

Thanks

**Comment by mmgoodnow** [comment]
Oh, someone had posted it in the discord and said they figured it out. I thought it was you

**Comment by ramzh** [comment]
Oh didn't know you had a discord. No unfortunately I have not, do you have any idea how the other user did?

**Comment by mmgoodnow** [comment]
They updated prowlarr

**Comment by ramzh** [comment]
Unfortunately, mine seems already on the latest:
![image](https://user-images.githubusercontent.com/20802073/211163495-5f29cac3-1def-4919-b7d5-d122e8bf97bd.png)

Is there anyway to get a verbose log of this?

**Comment by mmgoodnow** [comment]
Try doing `curl -v http://192.168.43.11:9696/12/api?apikey=?API` from inside the cross-seed container. What does the output look like?

**Comment by ramzh** [comment]
Command: `docker run -it --rm crossseed/cross-seed curl -v http://192.168.43.11:9696/12/api?apikey=?API`

Got the following: 
```
Configuration file created at /config/config.js
error: unknown command 'curl'
(add --help for additional information)
```

**Reply by mmgoodnow** [reply]
you have to shell into the container first. docker exec -it containerName sh

**Reply by ramzh** [reply]
Unfortunately, I am not sure how to. Every time I run the container, it closes itself 

`docker run --name cross-seed -it crossseed/cross-seed`

```
Configuration file created at /config/config.js
root@pc:~# docker ps -l
CONTAINER ID   IMAGE                  COMMAND                  CREATED         STATUS                     PORTS     NAMES
77846f57786b   crossseed/cross-seed                           8 seconds ago   Exited (1) 6 seconds ago             cross-seed
```

**Reply by mmgoodnow** [reply]
hmm, that's a good point. I think we need to override the entry point: 

```
sudo docker run -it --rm --entrypoint sh crossseed/cross-seed
```
(from https://phoenixnap.com/kb/docker-run-override-entrypoint)

That should get you a shell inside the container. From there you can run the curl command

**Reply by ramzh** [reply]
Got the output:
```
# curl -v http://192.168.43:11:9696/12/api?apikey=?APIKEY
* Expire in 0 ms for 6 (transfer 0x55923652f0f0)
*   Trying 192.168.43:11...
* TCP_NODELAY set
* Expire in 200 ms for 4 (transfer 0x55923652f0f0)
* Connected to 192.168.43:11 (192.168.43:11) port 9696 (#0)
> GET /12/api?apikey=?APIKEY HTTP/1.1
> Host: 192.168.43:11:9696
> User-Agent: curl/7.64.0
> Accept: */*
>
< HTTP/1.1 401 Unauthorized
< Content-Length: 0
< Date: Sat, 07 Jan 2023 20:30:02 GMT
< Server: Kestrel
<
* Connection #0 to host 192.168.43:11 left intact
```

Prowlarr has now an auth system, I put a bypass from local address but not sure if it's working.

**Reply by mmgoodnow** [reply, marked as answer]
Is the problem that you have a ? before your api key?

**Reply by ramzh** [reply]
I got it now
```
# curl -v http://192.168.43:11:9696/12/api?apikey=APIKEY
* Expire in 0 ms for 6 (transfer 0x558b526e20f0)
*   Trying 192.168.43:11...
* TCP_NODELAY set
* Expire in 200 ms for 4 (transfer 0x558b526e20f0)
* Connected to 192.168.43:11 (192.168.43:11) port 9696 (#0)
> GET /12/api?apikey=APIKEY HTTP/1.1
> Host: 192.168.43:11:9696
> User-Agent: curl/7.64.0
> Accept: */*
>
< HTTP/1.1 200 OK
< Content-Length: 95
< Content-Type: application/rss+xml
< Date: Sat, 07 Jan 2023 20:38:26 GMT
< Server: Kestrel
< Cache-Control: max-age=31536000, public
< Last-Modified: Tue, 03 Jan 2023 00:21:01 GMT
< X-Application-Version: 1.0.1.2220
<
<?xml version="1.0" encoding="UTF-8"?>
* Connection #0 to host 192.168.43.11 left intact
```

**Reply by mmgoodnow** [reply]
seems like that was it then. Does it work when you run cross-seed now (with those `?`s before the api keys removed?)

**Reply by ramzh** [reply]
Yup working! Thanks so much!

**Comment by LepaBrenaa** [comment]
I have the same issue.

I did everything as was described in [docs](https://www.cross-seed.org/docs/basics/getting-started).

I use this on my seedbox (USB provider).
Prowlarr is also set up correctly, and configured with 4 trackers.
Search via Prowlarr's web ui works perfectly.

The issue is when I run `cross-seed search`... I get the following message:
![image](https://user-images.githubusercontent.com/48409293/220906869-1ff9d551-a9b6-4acf-b947-c61f3ea93059.png)


Here is how the `torznab` section in `config.js` looks like:
```
torznab: [
        "https://boltok.solaris.usbx.me/prowlarr/3/api?KEY",
	    "https://boltok.solaris.usbx.me/prowlarr/4/api?apikey=KEY",
	    "https://boltok.solaris.usbx.me/prowlarr/1/api?apikey=KEY",
	    "https://boltok.solaris.usbx.me/prowlarr/2/api?apikey=KEY"
    ],
```

**Reply by spyhunter493** [reply]
try restarting prowlarr, failing that removing and re-adding the indexer from prowlarr fixed it for me

**Comment by AN1MATEK** [comment]
Did you guys ever got a solution? I am stuck here too. I tripled checked the API Key and restarted Prowlarr.

**Comment by roirraWedorehT** [comment]
I'm having this issue too.  I'm new to cross-seed, using it with Prowlarr via the npm method on Windows 10.  Two of the four private trackers indicate "Ignoring indexer that doesn't support searching".  I presume the remaining two work correctly - I just have less interest in cross-seeding from one of the two as it's less likely to match anywhere else, and if I'm going to test cross-seed, I'm much more interested in crossing to the two that aren't working.

I tried the earlier advice to restart Prowlarr, and I also removed those indexers from Prowlarr and re-added them, with no change.

I'm able to search and get results from those trackers manually just fine through Prowlarr.

Edit:  I'm using the latest Prowlarr (1.3.2.3006).

**Comment by pelican12383** [comment]
I ran into this issue as well, and found a workaround (for my issue, at least). My issue has to do with disabled indexers in prowlarr:

It looks like cross-seed requires the prowlarr indexer to be enabled on first use. Even if you run cross-seed with a disabled indexer, and then enable the indexer afterwards, cross-seed will still show that same warning and not use the indexer.

**Edit**: as pointed out in https://github.com/cross-seed/cross-seed/issues/440: There should be no reason to disable indexers in prowlarr in order to prevent other *arr clients from using them. After properly setting up the sync profiles in prowlarr, my issue is resolved.

~~On the other hand, if you start off with the indexer enabled, run cross-seed, and then disable the indexer, cross-seed seems to be OK with the indexer being disabled on subsequent runs (I gather that it's using some data saved in cross-seed.db for this).~~

~~Steps to run with disabled indexers:~~
~~1. Delete cross-seed.db **(or move/rename it if you don't want to wipe your data !!!)**
2. Enable the indexers in prowlarr (no need to delete/re-add them in prowlarr, or restart prowlarr)
3. Run cross-seed once
4. If desired, disable the indexers
5. cross-seed should continue using the disabled indexers~~

~~Alternatively, you could delete/re-add the indexers in prowlarr (keeping them enabled, at least for the first cross-seed run), and then you wouldn't need to delete cross-seed.db.~~

~~I've added a feature-request here to allow using disabled prowlarr indexers: https://github.com/cross-seed/cross-seed/issues/440~~

**Reply by salilkansal** [reply]
The issue indeed is that disabled indexer in prowlarr is not searchable via cross seed. Even after enabling it, looks like cross seed has some cache for the tracker which is not updated. I removed the entire cache, db and then enabled the tracker in prowlarr,  restarted cross seed and it worked now.

**Comment by mmgoodnow** [comment]
#452 should help here.
- when torznab requests fail, it logs the failure reason (use `--verbose` option to see first 1000 chars of the response body)
- will retry indexers that have failed to get caps in the past

going to merge it in soon.
